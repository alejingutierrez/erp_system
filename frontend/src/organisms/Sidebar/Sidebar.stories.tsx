import type { Meta, StoryObj } from '@storybook/react';
import { Sidebar } from './Sidebar';

const meta: Meta<typeof Sidebar> = {
  title: 'Organisms/Sidebar',
  component: Sidebar,
  tags: ['autodocs'],
  argTypes: {
    collapsed: { control: 'boolean' },
    navItems: { control: 'object' },
    onNavigate: { action: 'navigate' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

const navItems = [
    { label: 'Panel', iconName: 'LayoutDashboard', path: '/panel' },
    { label: 'Productos & Catálogo', iconName: 'Package', path: '/productos' },
    { label: 'Inventario', iconName: 'Warehouse', path: '/inventario' },
    { label: 'Compras & Abastecimiento', iconName: 'ShoppingCart', path: '/compras' },
    { label: 'Ventas & Pedidos', iconName: 'ShoppingBag', path: '/ventas' },
    { label: 'CRM & Atención', iconName: 'Users', path: '/crm' },
    { label: 'Marketing & Campañas', iconName: 'Megaphone', path: '/marketing' },
    { label: 'CDP & Audiencias', iconName: 'Target', path: '/cdp' },
    { label: 'Logística & Fulfillment', iconName: 'Truck', path: '/logistica' },
    { label: 'Finanzas & Contabilidad', iconName: 'Landmark', path: '/finanzas' },
    { label: 'Analytics & BI', iconName: 'AreaChart', path: '/analytics' },
    { label: 'Integraciones', iconName: 'Plug', path: '/integraciones' },
    { label: 'Configuración', iconName: 'Settings', path: '/configuracion' },
    { label: 'Ayuda & Soporte', iconName: 'HelpCircle', path: '/ayuda' },
];

export const Default: Story = {
  args: {
    navItems,
  },
};

export const Collapsed: Story = {
  args: {
    ...Default.args,
    collapsed: true,
  },
};

export const ActiveItem: Story = {
    args: {
      ...Default.args,
      activePath: '/productos',
    },
  };
