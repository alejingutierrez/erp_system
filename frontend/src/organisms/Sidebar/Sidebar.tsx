import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';
import { cn } from '@/lib/utils';
import { Icon } from '@/atoms/Icon';
import { NavItem, type NavItemProps } from '@/molecules/NavItem';

const sidebarVariants = cva(
  'flex flex-col gap-2 p-2 bg-background text-foreground transition-all duration-300',
  {
    variants: {
      collapsed: {
        true: 'w-16',
        false: 'w-64',
      },
    },
    defaultVariants: {
      collapsed: false,
    },
  },
);

export interface SidebarProps
  extends React.HTMLAttributes<HTMLElement>,
    VariantProps<typeof sidebarVariants> {
  navItems: NavItemProps[];
  onNavigate?: (path: string) => void;
  activePath?: string;
}

export const Sidebar = React.forwardRef<HTMLElement, SidebarProps>(
  ({ className, collapsed, navItems, onNavigate, activePath, ...props }, ref) => {
    return (
      <aside
        ref={ref}
        className={cn(sidebarVariants({ collapsed }), className)}
        {...props}
      >
        <nav className="flex flex-col gap-2">
            {navItems.map((item) => (
            <NavItem
                key={item.label}
                {...item}
                active={activePath === item.path}
                onClick={() => onNavigate?.(item.path || item.label)}
                label={collapsed ? '' : item.label}
            />
            ))}
        </nav>
      </aside>
    );
  },
);

Sidebar.displayName = 'Sidebar';
export { sidebarVariants };
