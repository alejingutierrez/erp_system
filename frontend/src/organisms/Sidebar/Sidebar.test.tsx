import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Sidebar } from './Sidebar';

const navItems = [
    { label: 'Panel', iconName: 'LayoutDashboard', path: '/panel' },
    { label: 'Productos & Catálogo', iconName: 'Package', path: '/productos' },
    { label: 'Inventario', iconName: 'Warehouse', path: '/inventario' },
];

describe('Sidebar', () => {
    it('should render correctly', () => {
        const { container } = render(<Sidebar navItems={navItems} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with no accessibility violations', async () => {
        const { container } = render(<Sidebar navItems={navItems} />);
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });

    it('should render all nav items', () => {
        render(<Sidebar navItems={navItems} />);
        expect(screen.getByText('Panel')).toBeInTheDocument();
        expect(screen.getByText('Productos & Catálogo')).toBeInTheDocument();
        expect(screen.getByText('Inventario')).toBeInTheDocument();
    });

    it('should not render labels when collapsed', () => {
        render(<Sidebar navItems={navItems} collapsed />);
        expect(screen.queryByText('Panel')).not.toBeInTheDocument();
        expect(screen.queryByText('Productos & Catálogo')).not.toBeInTheDocument();
        expect(screen.queryByText('Inventario')).not.toBeInTheDocument();
    });
});
