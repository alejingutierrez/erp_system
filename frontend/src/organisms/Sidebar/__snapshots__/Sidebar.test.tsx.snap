// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > should render correctly 1`] = `
<aside
  class="flex flex-col gap-2 p-2 bg-background text-foreground transition-all duration-300 w-64"
>
  <nav
    class="flex flex-col gap-2"
  >
    <button
      class="flex items-center gap-2 w-full rounded-md px-3 py-2 text-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-secondary text-foreground hover:bg-muted"
      path="/panel"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-layout-dashboard inline-block align-middle w-5 h-5"
        fill="none"
        focusable="false"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="9"
          rx="1"
          width="7"
          x="3"
          y="3"
        />
        <rect
          height="5"
          rx="1"
          width="7"
          x="14"
          y="3"
        />
        <rect
          height="9"
          rx="1"
          width="7"
          x="14"
          y="12"
        />
        <rect
          height="5"
          rx="1"
          width="7"
          x="3"
          y="16"
        />
      </svg>
      <span
        class="font-sans text-lg font-normal flex-1 text-left"
      >
        Panel
      </span>
    </button>
    <button
      class="flex items-center gap-2 w-full rounded-md px-3 py-2 text-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-secondary text-foreground hover:bg-muted"
      path="/productos"
      type="button"
    >
      <span
        class="font-sans text-lg font-normal flex-1 text-left"
      >
        Productos & Cat√°logo
      </span>
    </button>
    <button
      class="flex items-center gap-2 w-full rounded-md px-3 py-2 text-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-secondary text-foreground hover:bg-muted"
      path="/inventario"
      type="button"
    >
      <span
        class="font-sans text-lg font-normal flex-1 text-left"
      >
        Inventario
      </span>
    </button>
  </nav>
</aside>
`;
